python/llm/src/ipex_llm/transformers/models/llama.py

```diff
@@ -639,11 +639,18 @@ def llama_attention_forward_4_31_original(
         attn_weights = None
     else:
         # otherwise, use native attention
-        attn_output, attn_weights = native_sdp(query_states, key_states, value_states,
+        if query_states.dtype == torch.float16 and query_states.shape[2] >= 6800:
```

Comment:
also need to add the condition if don't return attn_weights?

Reply:
fixed.